# HG changeset patch
# User Ben Abbott <bpabbott@mac.com>
# Date 1286655940 14400
#      Sat Oct 09 16:25:40 2010 -0400
# Node ID d4619eb6ef8e1e6df594eedf5168a003208fcf95
# Parent  add5beb3b845279b7c9a86ab7d7f0dc2dc4daae1
__go_draw_axes__.m: Enable linetypes for gnuplot.

diff --git a/scripts/ChangeLog b/scripts/ChangeLog
--- a/scripts/ChangeLog
+++ b/scripts/ChangeLog
@@ -1,8 +1,12 @@
+2010-10-09  Ben Abbott <bpabbott@mac.com>
+
+	* plot/__go_draw_axes__.m: Enable linetypes for gnuplot.
+
 2010-10-09  John W. Eaton  <jwe@octave.org>
 
 	* statistics/distributions/geornd.m,
 	* statistics/distributions/hygecdf.m,
 	* statistics/distributions/hygeinv.m,
 	* statistics/distributions/poissrnd.m, 
 	* statistics/distributions/wblrnd.m:
 	Use || instead of | and && instead of & in IF conditions
diff --git a/scripts/plot/__go_draw_axes__.m b/scripts/plot/__go_draw_axes__.m
--- a/scripts/plot/__go_draw_axes__.m
+++ b/scripts/plot/__go_draw_axes__.m
@@ -1637,28 +1637,19 @@ function style = do_linestyle_command (o
       case "-."
         lt = "6";
       case "none"
         lt = "";
       otherwise
         lt = "";
     endswitch
 
-    ## FIXME -- linetype is currently broken, since it disables the
-    ## gnuplot default dashed and solid linestyles with the only
-    ## benefit of being able to specify '--' and get a single sized
-    ## dashed line of identical dash pattern for all called this way.
-    ## All dash patterns are a subset of "with lines" and none of the
-    ## lt specifications will correctly propagate into the x11 terminal
-    ## or the print command.   Therefore, it is currently disabled in
-    ## order to allow print (..., "-dashed") etc. to work correctly.
-
-    ##    if (! isempty (lt))
-    ##      fprintf (plot_stream, " linetype %s", lt);
-    ##    endif
+    if (! isempty (lt))
+      fprintf (plot_stream, " linetype %s", lt);
+    endif
 
   else
     lt = "";
   endif
   if (! isempty (errbars))
     found_style = true;
   endif
 
@@ -1675,17 +1666,17 @@ function style = do_linestyle_command (o
 
   sidx = 1;
   if (isempty (errbars))
     if (isempty (lt))
       style {sidx} = "";
     else
       style {sidx} = "lines";
     endif
-    
+
     facesame = true;
     if (! isequal (pt, pt2) && isfield (obj, "markerfacecolor") 
         && !strncmp (obj.markerfacecolor, "none", 4))
       if (strncmp (obj.markerfacecolor, "auto", 4)
           || ! isnumeric (obj.markerfacecolor) 
           || (isnumeric (obj.markerfacecolor) 
               && isequal (color, obj.markerfacecolor)))
         if (! isempty (pt2))
